plugin/storage/badger/spanstore/reader.go

```diff
@@ -494,34 +513,34 @@ func validateQuery(p *spanstore.TraceQueryParameters) error {
 }
 
 // scanIndexKeys scans the time range for index keys matching the given prefix.
-func (r *TraceReader) scanIndexKeys(indexKeyValue []byte, startTimeMin time.Time, startTimeMax time.Time) ([][]byte, error) {
+func (r *TraceReader) scanIndexKeys(indexKeyValue []byte, plan *executionPlan) ([][]byte, error) {
 	indexResults := make([][]byte, 0)
 
-	startStampBytes := make([]byte, 8)
-	binary.BigEndian.PutUint64(startStampBytes, model.TimeAsEpochMicroseconds(startTimeMin))
-
 	err := r.store.View(func(txn *badger.Txn) error {
 		opts := badger.DefaultIteratorOptions
 		opts.PrefetchValues = false // Don't fetch values since we're only interested in the keys
+		opts.Reverse = true
 		it := txn.NewIterator(opts)
 		defer it.Close()
 
 		// Create starting point for sorted index scan
-		startIndex := make([]byte, len(indexKeyValue)+len(startStampBytes))
+		startIndex := make([]byte, len(indexKeyValue)+8+1)
+		startIndex[len(startIndex)-1] = 0xFF
 		copy(startIndex, indexKeyValue)
-		copy(startIndex[len(indexKeyValue):], startStampBytes)
+		copy(startIndex[len(indexKeyValue):], plan.startTimeMax)
 
-		timeMax := model.TimeAsEpochMicroseconds(startTimeMax)
-		for it.Seek(startIndex); scanFunction(it, indexKeyValue, timeMax); it.Next() {
+		for it.Seek(startIndex); scanFunction(it, indexKeyValue, plan.startTimeMin); it.Next() {
```

Comment:
Has changed from startTimeMax to Min - is that correct? Is it because of reversing the time order?

Reply:
Exactly. We finish the scan when the min timestamp is hit.